src/sql/src/plan.rs

```diff
@@ -433,6 +433,10 @@ impl Plan {
     /// falling into the `true` category.
     pub fn allowed_in_read_only(&self) -> bool {
         match self {
+            // These two set non-durable session variables, so are okay in
+            // read-only mode.
+            Plan::SetVariable(_) => true,
+            Plan::ResetVariable(_) => true,
             Plan::SetTransaction(_) => true,
```

Comment:
It might be a bit verbose but it could be useful to explicitly list out all variants of `Plan` instead of using the wildcard `_`. That would require us to consider each existing plan individually and also any plans that are added in the future.